cpp/src/io/parquet/page_enc.cu

```diff
@@ -1741,8 +1752,18 @@ dremel_data get_dremel_data(column_view h_col, rmm::cuda_stream_view stream)
     },
     stream);
 
-  column_offsets.device_to_host(stream, true);
-  column_ends.device_to_host(stream, true);
+  thrust::host_vector<size_type> column_offsets(nesting_levels.size() + 1);
+  CUDA_TRY(cudaMemcpyAsync(column_offsets.data(),
+                           d_column_offsets.data(),
+                           d_column_offsets.size() * sizeof(size_type),
+                           cudaMemcpyDeviceToHost,
+                           stream.value()));
+  thrust::host_vector<size_type> column_ends(nesting_levels.size() + 1);
+  CUDA_TRY(cudaMemcpyAsync(column_ends.data(),
+                           d_column_ends.data(),
+                           d_column_ends.size() * sizeof(size_type),
+                           cudaMemcpyDeviceToHost,
+                           stream.value()));
```

Comment:
IIUC, `cudaMemcpyAsync` is actually [synchronous for Device to Host copies](https://docs.nvidia.com/cuda/cuda-runtime-api/api-sync-behavior.html#api-sync-behavior).

Reply:
Yes, they are for pageable host memory. But if we switch at some point to pinned host memory (which I imagine is likely), then there will be a hard to find bug in this code. So I suggest adding a `stream.synchronize()` here since it's not likely to hurt given these calls are currently synchronous.